src/reporters/base.ts

```diff
@@ -66,26 +66,7 @@ export default class BaseReporter {
     this.fd = options.fd || process.stdout.fd;
     this.stream = new SonicBoom({ fd: this.fd, sync: true });
     this._registerListeners();
-    this.runner.on('end', () => this.close());
-  }
-
-  close() {
-    if (this.fd <= 2) {
-      /**
-       * For stdout/stderr we destroy stream once data is written and run
-       * it as the last listener giving enough room for
-       * other reporters to write to stream
-       */
-      setImmediate(() => this.stream.end());
-    } else {
-      /**
-       * If the user has passed a custom FD we don't close the FD, but we do flush it
-       * to give them more control. This is important because FDs should only be closed
-       * once, and the primary use case for custom FDs is being called by heartbeat, which
-       * closes the FD after the process exits
-       */
-      this.stream.flush();
-    }
+    this.runner.on('end', () => this.stream.flush());
```

Comment:
We don't need to take care of closing the stream as the process exists automatically which closes the underlying stream once program runs to completion. This is better than figuring out when to close and prematurely closing it than leaving it which does not harm.